src/client/builtin_scene/materials/gaussian_splatting.cpp

```diff
@@ -0,0 +1,113 @@
+#include <memory>
+#include <vector>
+#include <client/graphics/webgl_context.hpp>
+#include <client/graphics/webgl_program.hpp>
+#include <client/graphics/webgl_buffer.hpp>
+#include <common/debug.hpp>
+
+#include "./gaussian_splatting.hpp"
+#include "../meshes.hpp"
+
+namespace builtin_scene::materials
+{
+  bool GaussianSplattingMaterial::initialize(std::shared_ptr<client_graphics::WebGL2Context> glContext,
+                                             std::shared_ptr<client_graphics::WebGLProgram> program)
+  {
+    if (TR_UNLIKELY(!Material::initialize(glContext, program)))
+      return false;
+
+#define LOAD_UNIFORM_LOCATION(name)                          \
+  {                                                          \
+    auto loc = glContext->getUniformLocation(program, name); \
+    if (loc.has_value())                                     \
+    {                                                        \
+      uniforms_.emplace(name, loc.value());                  \
+    }                                                        \
+  }
+    // Load uniforms for improved 3DGS rendering
+    LOAD_UNIFORM_LOCATION("renderSize");
+    LOAD_UNIFORM_LOCATION("maxStdDev");
+    LOAD_UNIFORM_LOCATION("minAlpha");
+    LOAD_UNIFORM_LOCATION("maxPixelRadius");
+    LOAD_UNIFORM_LOCATION("clipXY");
+    // Note: viewMatrix and projectionMatrix are handled automatically by WebGL context
+#undef LOAD_UNIFORM_LOCATION
+
+    // Set default values for 3DGS parameters
+    auto renderSizeOpt = glContext->getUniformLocation(program, "renderSize");
+    if (renderSizeOpt.has_value())
+      glContext->uniform2f(renderSizeOpt.value(), 1600.0f, 900.0f); // Default render size
```

Comment:
Set the render size from the drawing buffer size in the builtin_scene's WebGLContext.